R/utils-json.R

```diff
@@ -169,47 +169,74 @@ getApprovals <- function(data, chain_nm, legend_nm, appr_var_all, month=NULL, po
   approvals_all <- list()
   
   if(approvalsAtBottom==FALSE) {
-    for(approval in appr_type){
-      appr_var <- appr_var_all[which(appr_type == approval)]
-      
-      if(subsetByMonth){
-        points <- subsetByMonth(getTimeSeries(data, chain_nm), month)
-      } else {
-        points <- data[[chain_nm]][['points']]
-        points$time <- formatDates(points[['time']], type=NA)
-      }
-      
-      appr_dates <- getApprovalDates(data, chain_nm, approval)
-      date_index <- apply(appr_dates, 1, function(d, points){
-        which(points$time >= d[1] & points$time <= d[2])}, 
-        points=points)
-      
-      if(is.list(date_index)){
-        date_index_list <- date_index
-      } else {
-        date_index_list <- list(date_index)
-      }
+    approved_dates <- getApprovalDates(data, chain_nm, "Approved")
+    review_dates <-getApprovalDates(data, chain_nm, "In Review")
+    working_dates <- getApprovalDates(data, chain_nm, "Working")
       
+    if(subsetByMonth){
+      points <- subsetByMonth(getTimeSeries(data, chain_nm), month)
+      points_no_times <- points
+      points_no_times$time <- as.POSIXct(strptime(points_no_times$time, "%F"))
+    } else {
+      points <- data[[chain_nm]][['points']]
+      points_no_times <- points
+      points_no_times$time <- as.POSIXct(strptime(points_no_times[['time']], "%F"))
+    }
+
+    working_index<- apply(working_dates, 1, function(d, points){
+      which(points$time >= d[1] & points$time <= d[2])}, 
+      points=points_no_times)
+
+    review_index<- apply(review_dates, 1, function(d, points){
+      which(points$time >= d[1] & points$time <= d[2])}, 
+      points=points_no_times)
+
+    approved_index<- apply(approved_dates, 1, function(d, points){
+      which(points$time >= d[1] & points$time <= d[2])}, 
+      points=points_no_times)
+
+    if(is.list(working_index)){
```

Comment:
I would recommend not using `is.list()`. It's tricky -- data.frames are technically "lists" under the hood. Try `class(working_index) == "list")` instead.

Trickiness:
```r
ex <- data.frame(a=1,b=2)
is.list(ex)
[1] TRUE
```